### PR TITLE
Add an XmlCollection type, and update Styles to use it

### PR DESCRIPTION
**Pull request by kylegibson**
Currently Styles has to implement a collection-like interface. Abstract this implementation into XmlModel and define a XmlCollection type so other elements can benefit from it. 

This depends on #127

### CODE REVIEW

File: pydocx/models.py [hunk 1]
```diff
@@ -43,6 +46,36 @@ def __init__(self, name=None, default=None, type=None, attrname=None):
...
 
+class XmlCollection(XmlField):
+    '''
+    Represents a collection of elements. To define a field of this type,
+    a mapping dictionary must be passed in specifying the name of the child
+    elements and a callable handler for each.
+
+    Example:
+
+    class ParkingLot(XmlModel):
```
Comment (winhamwr) [inline pydocx/models.py]: :+1:
I really like this example!

File: pydocx/models.py [hunk 2]
```diff
@@ -142,14 +182,42 @@ def child_handler(child):
...
-                handler = child_handlers.get(child.tag, None)
-                if callable(handler):
-                    kwargs[field_name] = handler(child)
+        # Build a mapping of tag names to collections
+        collection_member_to_collections = defaultdict(list)
+        for field_name, field in collections.items():
+            for tag_name in field.name_to_type_map.keys():
+                collection_member_to_collections[tag_name].append(field_name)
+
+        if element:
```
Comment (winhamwr) [inline pydocx/models.py]: Why do we want this conditional? `for child in element` should do this implicitly, right?

Comment (kylegibson) [inline pydocx/models.py]: Not if element is `None`, which it could be.

File: tests/openxml/wordprocessing/test_run_properties.py [hunk 3]
```diff
@@ -0,0 +1,73 @@
+# coding: utf-8
```
Comment (winhamwr) [inline tests/openxml/wordprocessing/test_run_properties.py]: :+1: to `RunProperties` having their own test module.

File: tests/util/test_zip.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    unicode_literals,
+)
+
+from unittest import TestCase
+
+from pydocx.exceptions import MalformedDocxException
+from pydocx.util.zip import ZipFile, BytesIO
+
+
+class ZipFileTestCase(TestCase):
```
Comment (winhamwr) [inline tests/util/test_zip.py]: I was a bit surprised to see this test added in this PR.

**Comment by winhamwr** [comment]
Looks good!

File: tests/util/test_zip.py (hunk 4 above)
Comment (kylegibson) [inline tests/util/test_zip.py]: I was waiting on review, so I added a few missing test cases to help with code coverage.

**Pull request merged**